Gem/CMakeLists.txt

```diff
@@ -49,29 +49,29 @@ ly_add_target(
 )
 
 ly_add_target(
-    NAME ROS2-Gem-Demo ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAME RobotVacuumSample ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
     NAMESPACE Gem
     FILES_CMAKE
-        ros2-gem-demo_shared_files.cmake
-        ${pal_dir}/ros2-gem-demo_shared_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+        robot_vacuum_sample_shared_files.cmake
+        ${pal_dir}/robot_vacuum_sample_shared_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
     INCLUDE_DIRECTORIES
         PUBLIC
             Include
     BUILD_DEPENDENCIES
         PRIVATE
-            Gem::ROS2-Gem-Demo.Static
+            Gem::RobotVacuumSample.Static
             AZ::AzCore
 )
 
-# if enabled, ROS2-Gem-Demo is used by all kinds of applications
-ly_create_alias(NAME ROS2-Gem-Demo.Builders NAMESPACE Gem TARGETS Gem::ROS2-Gem-Demo)
-ly_create_alias(NAME ROS2-Gem-Demo.Tools    NAMESPACE Gem TARGETS Gem::ROS2-Gem-Demo)
-ly_create_alias(NAME ROS2-Gem-Demo.Clients  NAMESPACE Gem TARGETS Gem::ROS2-Gem-Demo)
-ly_create_alias(NAME ROS2-Gem-Demo.Servers  NAMESPACE Gem TARGETS Gem::ROS2-Gem-Demo)
+# if enabled, RobotVacuumSample is used by all kinds of applications
+ly_create_alias(NAME RobotVacuumSample.Builders NAMESPACE Gem TARGETS Gem::RobotVacuumSample)
+ly_create_alias(NAME RobotVacuumSample.Tools    NAMESPACE Gem TARGETS Gem::RobotVacuumSample)
+ly_create_alias(NAME RobotVacuumSample.Clients  NAMESPACE Gem TARGETS Gem::RobotVacuumSample)
+ly_create_alias(NAME RobotVacuumSample.Servers  NAMESPACE Gem TARGETS Gem::RobotVacuumSample)
 
 ################################################################################
 # Gem dependencies
 ################################################################################
 
 # Enable the specified list of gems from GEM_FILE or GEMS list for this specific project:
-ly_enable_gems(PROJECT_NAME ROS2-Gem-Demo GEM_FILE enabled_gems.cmake)
+ly_enable_gems(PROJECT_NAME RobotVacuumSample GEM_FILE enabled_gems.cmake)
```

Comment:
Can you update this ly_enable_gems logic to the following to help with renaming the project in the future?
```suggestion
# Query the project name from the nearest project.json file in a directory at or above
# the current one.
# This gem is the project gem and therefore should be part of the project that is using it
o3de_find_ancestor_project_root(project_path project_name "${CMAKE_CURRENT_SOURCE_DIR}")


# If the project name could not be queried from a project.json file, then fallback
# to using the name of the project provided when the project template was instantiated
if (NOT project_name)
    set(project_name RobotVacuumSample)
endif()


# Enable the specified list of gems from GEM_FILE or GEMS list for this specific project:
ly_enable_gems(PROJECT_NAME ${project_name} GEM_FILE enabled_gems.cmake)
```

Reference: https://github.com/o3de/o3de/blob/development/Templates/DefaultProject/Template/Gem/CMakeLists.txt#L90-L102